dask/dataframe/io/parquet/arrow.py

```diff
@@ -625,6 +654,181 @@ def _construct_parts(
     return parts, stats
 
 
+#
+#  PyArrow Dataset API [PyArrow>=1.0.0] - Private Helper Functions
+#
+
+
+def _get_all_partition_keys(ds):
+    pkeys = defaultdict(list)
+    for file_frag in ds.get_fragments():
+        keys = pa_ds._get_partition_keys(file_frag.partition_expression)
+        for k, v in keys.items():
+            if v not in pkeys[k]:
+                pkeys[k].append(v)
+    return pkeys
+
+
+def _collect_pyarrow_dataset_frags(ds, filters):
+    partition_names = []
+    partition_obj = []
+    partition_keys = {}
```

Comment:
For understanding what is happening, it might be useful to add some comments what those three different object exactly are

(it might be this is already clear from / documented in the rest of the implementation that is not in the diff, though)